From: Thorsten Blum <blum@kernel.org>
Subject: init: Fix early boot crash with bare hostname parameter
Date: Wed, 26 Aug 2026 12:09:03 +0200

When a bare hostname parameter is specified on the kernel command line
without the '=' separator, early parameter parsing passes NULL to
early_hostname(), which dereferences it in strscpy() and can crash the
system during early boot.

Reject NULL values in early_hostname() and return -EINVAL instead.

Link: https://lore.kernel.org/20260826100904.296151-2-blum@kernel.org
Fixes: 5a704629f2c1 ("init: add "hostname" kernel parameter")
Signed-off-by: Thorsten Blum <blum@kernel.org>
Cc: Dan Moulding <dmoulding@me.com>
Cc: <stable@vger.kernel.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 init/version.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/init/version.c~init-fix-early-boot-crash-with-bare-hostname-parameter
+++ a/init/version.c
@@ -23,6 +23,9 @@ static int __init early_hostname(char *a
 	size_t maxlen  = bufsize - 1;
 	ssize_t arglen;
 
+	if (!arg)
+		return -EINVAL;
+
 	arglen = strscpy(init_uts_ns.name.nodename, arg, bufsize);
 	if (arglen < 0) {
 		pr_warn("hostname parameter exceeds %zd characters and will be truncated",
_
